decentraland/screens/EthereumSignUpScreen.js

```diff
@@ -13,31 +13,41 @@ import {
 import { Account } from "tasit-sdk";
 
 export class EthereumSignUpScreen extends React.Component {
-  // TODO: Use Gnosis Safe
   _onboarding = async () => {
     try {
-      const { setAccount } = this.props;
+      const { setAccount, setSetupInProgress } = this.props;
 
       // Note: The timeout for account creation is about ~10 secs.
       // See more: https://github.com/tasitlabs/tasit/issues/42
       const account = Account.create();
       setAccount(account);
 
       const { address: accountAddress } = account;
-      await fundAccount(accountAddress);
 
-      await approveManaSpending(account);
-      showInfo(`Account created and funded!`);
+      const onFundSuccess = async () => {
+        const onApprovalSuccess = () => {
+          setSetupInProgress(false);
+          showInfo(`Account created and funded!`);
+        };
+
+        approveManaSpending(account, onApprovalSuccess);
+      };
+
+      fundAccount(accountAddress, onFundSuccess);
     } catch (error) {
       showError(error);
     }
   };
 
   _onSignUp = () => {
-    showInfo(`Creating and funding account...`);
-
-    // Should run async but isn't when calling Account.create() or createFromPrivateKey()
-    this._onboarding();
+    const { setSetupInProgress } = this.props;
+    setSetupInProgress(true);
+    // Note: A trick to force `_onboarding()` function to running async
+    (async () => {})().then(() => {
```

Comment:
Interesting. Nice hack! We can continue to ponder any alternate strategies but this will serve us very well in the meantime.

Reply:
Resolving this